test/model/create.test.ts

```diff
@@ -29,3 +29,63 @@ test('creates a new entity with initial values', () => {
   })
   expect(exactUser).toHaveProperty('id', 'abc-123')
 })
+
+test('creates a new entity with initial values, including arrays', () => {
+  const db = factory({
+    user: {
+      id: primaryKey(datatype.uuid),
+      arrayProp: Array,
+    },
+  })
+
+  const exactUser = db.user.create({
+    id: 'abc-123',
+    arrayProp: [1, 2, 3],
+  })
+  expect(exactUser).toHaveProperty('id', 'abc-123')
+  expect(exactUser).toHaveProperty('arrayProp', [1, 2, 3])
+})
+
+test('creates a new entity with initial values, including nested objects', () => {
+  const db = factory({
+    user: {
+      id: primaryKey(datatype.uuid),
+      name: String,
+      info: {
+        firstName: String,
```

Comment:
We need a test case that ensures that relational properties can be declared in nested objects.

```js
{
  user: {
    billingAddress: {
      country: oneOf('country')
    }
  },
  country: { ... }
}
```

Reply:
wrote the test, it failed...working on it to make it pass :)

Reply:
I believe you'll need to analyze model properties recursively because of nested objects support. Should be a matter of introducing recursion to `parseModelDeclaration` or friends.

Reply:
@kettanaito the test is now passing (other ones are failing, currently having a look), altough the code can be further improved, cleaned, simplified.
Btw I pushed it to see if the direction is good :) 
I also got rid of `flat` library as you suggested.